### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by AlexTerhar**
icon is set, background will remain transparent, not changable, icon is green, Chanel is set 
If its possible, someone else can finish the pull

### CODE REVIEW

**Comment by frcroth** [comment]
So this is not yet complete?

**Comment by AlexTerhar** [comment]
it is complete. but some desired changes are not possible due to the android oreo version

File: app/src/main/java/app/wefridge/wefridge/MainActivity.kt [hunk 1]
```diff
@@ -30,6 +35,7 @@ class MainActivity : AppCompatActivity() {
         this.onSignInResult(res)
     }
 
+    @SuppressLint("StringFormatInvalid")
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MainActivity.kt:38]: why this?

File: app/src/main/java/app/wefridge/wefridge/MainActivity.kt [hunk 2]
```diff
@@ -38,6 +44,17 @@ class MainActivity : AppCompatActivity() {
...
         // Successfully signed in
         val user = FirebaseAuth.getInstance().currentUser ?: return authWall()
+        Log.d("FCM", "This is the user id: ${user.uid} ")
+        Firebase.messaging.subscribeToTopic(user.uid)
+            .addOnCompleteListener { task ->
+                var msg = "subscribed to:"+user.uid
+                if (!task.isSuccessful) {
+                    msg = "subscribe_failed"
+                }
+                Log.d(R.string.TAG_Main.toString(), msg)
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MainActivity.kt:54]:
```suggestion
                Log.d(R.string.TAB_Main.toString(), if (task.isSuccessful) "Subscribed to ${user.uid}" else "Failed to subscribe")
```

File: app/src/main/java/app/wefridge/wefridge/MainActivity.kt [hunk 3]
```diff
@@ -38,6 +44,17 @@ class MainActivity : AppCompatActivity() {
         }
         // Successfully signed in
         val user = FirebaseAuth.getInstance().currentUser ?: return authWall()
+        Log.d("FCM", "This is the user id: ${user.uid} ")
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MainActivity.kt:47]:
```suggestion
        Log.d("FCM", "User ${user.uid} successfully signed in")
```

File: app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import android.util.Log
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+
+const val channelId="notification_channel"
+const val channelName="app.wefridge.wefridge"
+class MyFirebaseMessagingService : FirebaseMessagingService() {
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt:20]: Change name. Something like WeFridgeMessagingService would be better.

File: app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * Called when message is received.
+     *
+     * @param remoteMessage Object representing the message received from Firebase Cloud Messaging.
+     */
+    // [START receive_message]
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt:26]: What do these comments mean?

File: app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    }
+    // [END on_new_token]
+
+    /**
+     * Create and show a simple notification containing the received FCM message.
+     *
+     * @param title,message FCM message body received.
+     */
+    @SuppressLint("RemoteViewLayout")
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt:57]: Why this?

File: app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    fun getRemoteView(title: String, message: String): RemoteViews {
+
+        val remoteView= RemoteViews("app.wefridge.wefridge",R.layout.notification)
+        remoteView.setTextViewText(R.id.title_logo,title)
+        remoteView.setTextViewText(R.id.message,message)
+        remoteView.setImageViewResource(R.id.app_logo,R.drawable.ic_launcher_foreground)
+        return remoteView
+
+    }
+    @SuppressLint("UnspecifiedImmutableFlag")
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt:67]: Why this?

File: app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt (hunk 5 above)
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt:26]: What do these comments mean?

File: app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    private fun generateNotification(title:String, message: String) {
+        val intent = Intent(this, MainActivity::class.java)
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MyFirebaseMessagingService.kt:69]: Is the code of this method original or is there some source?

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -90,6 +90,11 @@
     <string name="settings_contact_email_hint">This is used, if someone wants to contact you about your shared items</string>
     <string name="avatar_description">Avatar</string>
     <string name="logout">Logout</string>
+    <string name="msg_token_fmt">FCM Token:</string>
+    <string name="TAG_Messservice">Messegingservice:</string>
+    <string name="TAG_Main">Main_Activity:</string>
+    <string name="default_notification_channel_id">app.wefridge.wefridge</string>
+    <string name="fcm_message">Hier könnte deine Werbung stehen</string>
```
Comment (frcroth) [inline app/src/main/res/values/strings.xml:97]: Unused I think

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -90,6 +90,11 @@
     <string name="settings_contact_email_hint">This is used, if someone wants to contact you about your shared items</string>
     <string name="avatar_description">Avatar</string>
     <string name="logout">Logout</string>
+    <string name="msg_token_fmt">FCM Token:</string>
+    <string name="TAG_Messservice">Messegingservice:</string>
+    <string name="TAG_Main">Main_Activity:</string>
```
Comment (frcroth) [inline app/src/main/res/values/strings.xml:95]: I don't think the TAGs need to be here

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -90,6 +90,11 @@
...
+    <string name="default_notification_channel_id">app.wefridge.wefridge</string>
```
Comment (frcroth) [inline app/src/main/res/values/strings.xml:96]: Also may not be the right place, just define this in the class I would say.

**Review by frcroth — CHANGES_REQUESTED** [review]

**Comment by AlexTerhar** [comment]
i made a new one with changes

**Pull request closed without merging**